### PR TITLE
[MRG] Fix `multigather` so that the output CSV contains all matches.

### LINKED ISSUE #2321: `multigather` only saves first match to CSV
when running:
```
sourmash multigather --query SRR606249-abund-10k.sig.zip --db out.txt -k 31 --threshold-bp=0
```
the resulting file `SRR606249.trim.fq.gz.csv` (which is based on the query signature `filename` property... ugh) only contains the first result.

this code is at fault in `commands.py::multigather`:
```
            w = None
            with FileOutputCSV(output_csv) as fp:
                for result in found:
                    if w is None:
                        w = result.init_dictwriter(fp)
                        result.write(w)
```
where the last line _should_ be deindented so that it is run every time through the loop.

### PR DESCRIPTION
**Pull request by ctb**
This PR fixes #2321 so that more than one output line is placed in the CSV. Oops!

It also adds a notification of what the CSV output file name is.

Last but not least, it supports `--output-dir` as a way to set the base path for all output files.

Fixes https://github.com/sourmash-bio/sourmash/issues/2321.

TODO:

- [x] add tests
- [x] make sure filename output behavior is documented
- [x] consider adding an option to have multigather save CSV results in some other way, like by md5 or ...something. - punted to https://github.com/sourmash-bio/sourmash/issues/2328

### CODE REVIEW

**Comment by ctb** [comment]
@sourmash-bio/devs @dkoslicki ready for review!

File: src/sourmash/cli/multigather.py [hunk 1]
```diff
@@ -83,6 +83,11 @@ def subparser(subparsers):
     )
     subparser.set_defaults(fail_on_empty_database=True)
 
+    subparser.add_argument(
```
Comment (dkoslicki) [inline src/sourmash/cli/multigather.py:86]: This is also appreciated: took me a minute to realize it just spit them out in the $CWD

**Review by dkoslicki — APPROVED** [review]
Looks great to me! When I have a chance, I'll probably give a shot at parallelizing this. I have plans for 1000's of `gather`s

File: src/sourmash/cli/multigather.py (hunk 1 above)
Comment (ctb) [reply src/sourmash/cli/multigather.py:86]: yep 😆

Comment (ctb) [reply src/sourmash/cli/multigather.py:86]: (regarding parallelizing - super easy with snakemake, I think, just ask!)

**Pull request merged**